lib/src/mock-server/proxy.ts

```diff
@@ -1,27 +1,38 @@
 import { Request, Response } from "express";
 import http from "http";
 import https from "https";
+import { ProxyConfig } from "./server";
 
 export function proxyRequest({
   incomingRequest,
   response,
-  protocol,
-  proxyBaseUrl
+  proxyConfig
 }: {
   incomingRequest: Request;
   response: Response;
-  protocol: "http" | "https";
-  proxyBaseUrl: string;
+  proxyConfig: ProxyConfig;
 }): void {
-  const requestHandler = protocol === "http" ? http : https;
+  console.log(proxyConfig);
+  const requestHandler = proxyConfig.isHttps ? https : http;
 
   const options = {
-    headers: incomingRequest.headers,
     method: incomingRequest.method,
-    path: incomingRequest.path
+    host: proxyConfig.host,
+    port:
+      proxyConfig.port === null
+        ? proxyConfig.isHttps
+          ? 443
+          : 80
+        : proxyConfig.port,
+    path: proxyConfig.path + incomingRequest.path,
+    headers: {
+      ...incomingRequest.headers,
+      host: proxyConfig.host
+    }
   };
+  console.log(options);
```

Comment:
**Question**: is it intentional to keep this log here?

Reply:
Oops, no. That wasn't intended. I'll remove them now.